### PR TITLE
[Merged by Bors] - feat: port Data.Finset.Prod

### PR DESCRIPTION
**Pull request by ChrisHughes24**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Finset/Prod.lean [hunk 1]
```diff
@@ -0,0 +1,448 @@
...
+section Prod
+
+variable {s s' : Finset α} {t t' : Finset β} {a : α} {b : β}
+
+/-- `product s t` is the set of pairs `(a, b)` such that `a ∈ s` and `b ∈ t`. -/
+protected def product (s : Finset α) (t : Finset β) : Finset (α × β) :=
+  ⟨_, s.nodup.product t.nodup⟩
+#align finset.product Finset.product
+
+--Porting note: Change notation from  "×ᶠ" to "×ᶠ" to avoid ambiguity
```
Comment (dupuisf) [inline Mathlib/Data/Finset/Prod.lean:47]: Is there a difference between those that I'm not seeing?

File: Mathlib/Data/Finset/Prod.lean [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+theorem filter_product_right (q : β → Bool) :
+    ((s ×ᶠ t).filter fun x : α × β => q x.2) = s ×ᶠ t.filter q := by
+  simpa using filter_product (fun _ : α => true) q
+#align finset.filter_product_right Finset.filter_product_right
+
+theorem filter_product_card (s : Finset α) (t : Finset β) (p : α → Bool) (q : β → Bool) :
+    ((s ×ᶠ t).filter fun x : α × β => (p x.1) == (q x.2)).card =
+      (s.filter p).card * (t.filter q).card +
+        (s.filter (not ∘ p)).card * (t.filter (not ∘ q)).card :=
+  by
```
Comment (dupuisf) [inline Mathlib/Data/Finset/Prod.lean:171]:
```suggestion
        (s.filter (not ∘ p)).card * (t.filter (not ∘ q)).card := by
```

**Review by dupuisf — COMMENTED** [review]
Looks good!

bors d+

File: Mathlib/Data/Finset/Prod.lean (hunk 1 above)
Comment (ChrisHughes24) [reply Mathlib/Data/Finset/Prod.lean:47]: Overzealous search and replace is responsible for this

**Pull request closed without merging**